parsl/tests/test_staging/test_elaborate_noop_file.py

```diff
@@ -0,0 +1,82 @@
+# this replaces the no-op file staging provider with a more
+# complicated no-op provider that launches (empty) tasks
+# for staging
+
+# the elaborate no-op provider is based around the structure
+# of the globus staging provider
+
+import logging
+import pytest
+
+import parsl
+
+from parsl import bash_app, python_app
+from parsl.config import Config
+from parsl.data_provider.files import File
+from parsl.executors.threads import ThreadPoolExecutor
+from parsl.tests.test_staging.staging_provider import NoOpTestingFileStaging
+from parsl.tests.utils import get_rundir
+
+logger = logging.getLogger(__name__)
+
+
+@bash_app
+def touch(filename, inputs=[], outputs=[]):
+    return "touch {}".format(filename)
+
+
+@python_app
+def app_test_in(file):
+    # does not need to do anything as this
+    # is just for exercising staging
+    pass
+
+
+@pytest.mark.local
+def test_regression_stage_out_does_not_stage_in():
+    no_stageout_config = Config(
+        executors=[
+            ThreadPoolExecutor(
+                label='local_threads',
+                storage_access=[NoOpTestingFileStaging(allow_stage_in=False)]
+            )
+        ],
+        run_dir=get_rundir()
+    )
+
+    parsl.load(no_stageout_config)
+
+    # test with no staging
+    touch("test.1", outputs=[File("test.1")]).result()
```

Comment:
It is not clear what test.1 vs test.2 does. Both invocations look the same?

Reply:
This should be addressed in 8566c4402  - I changed the invocation to test what the comment says.